website/app/styles/markdown/typography.scss

```diff
@@ -1,29 +1,29 @@
-@use "../typography/mixins.scss";
+@use "../typography/mixins";
 
 .doc-markdown-content {
 
   // HEADINGS
 
   > h1 {
```

Comment:
Would we be able to ensure that these tags are always direct children of the `.doc-markdown-content` parent element when the markdown is rendered to HTML?

Reply:
This is the challenge :) 

The reason for this way of writing CSS selectors is that we want to avoid leaking of the styles to the HDS components, in case they're embedded within the markdown.

In _theory_, we could imagine to _always_ make sure that the rendered markdown is wrapped in a container with class `doc-markdown-content` (potentially with `display: contents`).

In _practice_ what we will try to do is to leverage [this feature of Showdown](https://showdownjs.com/docs/tutorials/add-default-class-to-html/) that allows to assign custom classes to the HTML elements generated via markdown, in which case we can completely avoid the need of direct descendants and `doc-markdown-content` wrappers (there's a ticket for that: 
[HDS-823](https://hashicorp.atlassian.net/browse/HDS-823))

Reply:
Thanks for the background info. I understand you did it that way to avoid style leakage.

Would it be possible to render HDS components inside iframes as another possible solution to style leakage? Just curious as that's the way Storybook handles it.

Reply:
> Would it be possible to render HDS components inside iframes as another possible solution to style leakage? Just curious as that's the way Storybook handles it.

That was my first though! Hence my toying with the idea of a React application for the documentation, that would render "rich" markdown (via MDX) + Ember components (via iframe). 

The "problems" to overcome when using iframes are multiple though:
- each iframe should load its own JS files and CSS files; essentially, it would be a full blown Ember app - which means we should have a second web-app that renders the content of the iframe, but in that case how would we pass the "code" to the iframe? it would be a URL (not great)? or inline code that is rendered at runtime (but would it work in Ember? I did something similar with React, but I am not sure is possible with Ember)
- each frame should auto-resize to fit its content (and in that case, what if the content can overflow the bounding box, like a dropdown for example?)
- in a typical documentation page, we could have a hundred instances of a component, which means hundreds of iframes loading in parallel (probably not an issue from the bandwidth perspective, because of the cache, but JS should still need to be interpreted for each one of that instances) and hundreds of iframes being resized in parallel

If you want we can rubberduck a bit more this idea, and see if we can come up with solutions to the problems above.

> that's the way Storybook handles it

Yes, but in Storybook it's only one single large iframe, no?